### DISCUSSION TITLE
Meter per pixel

### DISCUSSION THREAD

**Discussion by pusia1** [pelicanmapping/osgearth#2423] [Support Q&A]
Hello.
Can I get meters per pixel with latitude and screen distance?

**Comment by gwaldron** [comment]
One approach is to pick two points that are one meter apart and convert them both into window coordinates.
* Take you lat/long point and convert it to ECEF coordinates (meters).
* Transform that point into window coordinates (pixels) by multiplying it by (view matrix * projection matrix * window matrix).
* Pick a second point 1 meter away from the first point and repeat.
* The distance between the 2 points in window coordinates is the number pixels per meter.
* Invert that and you have meters/pixel, more or less.

You can also refer to the algorithm in OSG that calculates an approximate pixel size for a bounding sphere:
https://github.com/openscenegraph/OpenSceneGraph/blob/master/include/osg/CullingSet#L174

Good luck.
